### ISSUE TITLE
clang 13.0.1: Assertion `declaresSameEntity(FD, LHS.Designator.Entries[PathLength] .getAsBaseOrMember() .getPointer())' failed.

### ISSUE THREAD

**Issue by lanodan** [llvm/llvm-project#54567]
- OS: Gentoo
- LLVM/Clang 13.0.1
- Source Tree: <https://www.webkitgtk.org/releases/webkitgtk-2.36.0.tar.xz> + <https://hacktivis.me/git/etc_portage.work/patches/net-libs/webkit-gtk/contextMenu-pointer_anchor-at-center.patch>
- Build log (includes clang backtrace): [net-libs:webkit-gtk-2.36.0-r200:20220326-100324.log](https://github.com/llvm/llvm-project/files/8355173/net-libs.webkit-gtk-2.36.0-r200.20220326-100324.log) (GTK4 target) 
[net-libs:webkit-gtk-2.36.0:20220326-100433.log](https://github.com/llvm/llvm-project/files/8355195/net-libs.webkit-gtk-2.36.0.20220326-100433.log) (GTK3 + libsoup2 target)
- [SpecializationConstant.cpp](https://github.com/llvm/llvm-project/files/8355183/SpecializationConstant.cpp.txt) (from GTK3 + libsoup2 target)

**Comment by DimitryAndric** [comment]
When you get an assertion, please attach the preprocessed source and shell script which is produced by clang. Otherwise it is very difficult to reproduce for other people. In this case:

```
/var/tmp/notmpfs/portage/net-libs/webkit-gtk-2.36.0-r200/temp/SpecializationConstant-f2bbbc.cpp
/var/tmp/notmpfs/portage/net-libs/webkit-gtk-2.36.0-r200/temp/SpecializationConstant-f2bbbc.sh
```

**Comment by DimitryAndric** [comment]
FWIW, this was fixed by 30baa5d2a450d5e302d8cba3fc7a26a59d4b7ae1 for #45224.
